backend/src/main/kotlin/database/PostgreRepository.kt

```diff
@@ -0,0 +1,164 @@
+        StoreTable.update({ StoreTable.id eq storeId }) {
+            it[name] = newName
+        }
+    }
+
+    override fun importSeasons(seasons: List<Season>): Unit = transaction(database) {
+        SeasonTable.batchInsert(seasons) { season ->
+            this[SeasonTable.half] = season.half.toString()
+            this[SeasonTable.year] = season.year.value
+        }
+    }
+
+    override fun getSeasons(): List<Season> = transaction(database) {
+        SeasonTable.selectAll().map {
+            Season(
+                half = it[SeasonTable.half].toSeasonHalf(),
+                year = Year.of(it[SeasonTable.year])
+            )
+        }
+    }
+
+    override fun updateSeasons(seasons: List<Season>): Unit = transaction(database) {
+        for (season in seasons) {
+            SeasonTable.update({ (SeasonTable.half eq season.half.toString()) and (SeasonTable.year eq season.year.value) }) {
+                it[half] = season.half.toString()
+                it[year] = season.year.value
+            }
+        }
+    }
+
+    override fun importStoreSeasons(map: List<Pair<Long, Season>>) {
+        for ((id, season) in map) {
+            try {
+                transaction(database) {
+                    StoreSeasonTable.insert {
+                        it[storeId] = id
+                        it[half] = season.half.toString()
+                        it[year] = season.year.value
+                    }
+                }
+            } catch (ex: SQLException) {
+                if (ex.cause?.message?.contains("fk_stores_seasons_storeid_id") == true) {
+                    println("Could not insert store season for Store id $id and season ${season.half} ${season.year} due to FK constraint. Will continue execution")
+                } else {
+                    throw ex
+                }
+            }
+        }
+    }
+
+    override fun getStoreSeasons(): List<Pair<Long, Season>> = transaction(database) {
+        val result = mutableListOf<Pair<Long, Season>>()
+        StoreSeasonTable.selectAll().map {
+            result.add(
+                Pair(
+                    it[StoreSeasonTable.storeId],
+                    Season(
+                        half = it[StoreSeasonTable.half].toSeasonHalf(),
+                        year = Year.of(it[StoreSeasonTable.year])
+                    )
+                )
+            )
+        }
+        return@transaction result
+    }
+
+    override fun deleteAll(): Unit = transaction(database) {
+        StoreTable.deleteAll()
+        SeasonTable.deleteAll()
+        StoreSeasonTable.deleteAll()
+    }
+
+    fun updateSchema() {
+        transaction(database) {
+            SchemaUtils.createMissingTablesAndColumns(StoreTable, SeasonTable, StoreSeasonTable)
+
+            TransactionManager.current()
+                .exec("""ALTER TABLE stores_seasons DROP CONSTRAINT IF EXISTS fk_stores_seasons;""")
```

Comment:
Why are you dropping a fkey?

Also, wont this affect every part of the application that could make use of the Configs' `repository`?

Reply:
As I have explained in the other issue: 

> JetBrains Exposed ORM has no way to programmatically deal with composite foreign keys.
I found this extremely unsatisfactory, but the only way I found to keep moving forward was by injecting some manual SQL commands to create his foreign key at database boot time.

This SQL running at boot time was breaking the app, as I was trying to add a FK that already existed. One way to work around this is to drop the FK and adding it again afterwards. I can live with this for a development app, but in a production environment It would be better to search for the FK, and only adding if it exists.
More context in this [stackoverlflow question](https://stackoverflow.com/questions/6801919/postgres-add-constraint-if-it-doesnt-already-exist)